psi4/driver/qcdb/libmintsbasisset.py

```diff
@@ -520,76 +520,86 @@ def build(molecule, shells):
-#TRIAL#
-#TRIAL#        #for label, basis_map in combined_atom_basis_shell.items():
-#TRIAL#        #    # sort the shells by angular momentum
-#TRIAL#        #    combined_atom_basis_shell[label][name] = sorted(combined_atom_basis_shell[label][name], key=lambda shell: she
-#TRIAL#
-#TRIAL#        # Molecule and parser prepped, call the constructor
-#TRIAL#        mol.set_basis_all_atoms(name, "CABS")
-#TRIAL#
-#TRIAL#        # Construct the grand BasisSet for mol
-#TRIAL#        basisset = BasisSet("CABS", mol, combined_atom_basis_shell)
-#TRIAL#
-#TRIAL#        # Construct all the one-atom BasisSet-s for mol's CoordEntry-s
-#TRIAL#        for at in range(mol.natom()):
-#TRIAL#            oneatombasis = BasisSet(basisset, at)
-#TRIAL#            oneatombasishash = hashlib.sha1(oneatombasis.print_detail(numbersonly=True).encode('utf-8')).hexdigest()
-#TRIAL#            mol.set_shell_by_number(at, oneatombasishash, role="CABS")
-#TRIAL#        mol.update_geometry()  # re-evaluate symmetry taking basissets into account
-#TRIAL#
-#TRIAL#        text = """   => Creating Basis Set <=\n\n"""
-#TRIAL#        text += """    Role: %s\n""" % (fitroles)
-#TRIAL#        text += """    Keyword: %s\n""" % (keys)
-#TRIAL#        text += """    Name: %s\n""" % (name)
-#TRIAL#
-#TRIAL#        if returnBasisSet:
-#TRIAL#            print(text)
-#TRIAL#            return basisset
-#TRIAL#        else:
-#TRIAL#            bsdict = {}
-#TRIAL#            bsdict['message'] = text
-#TRIAL#            bsdict['name'] = basisset.name
-#TRIAL#            bsdict['puream'] = int(basisset.has_puream())
-#TRIAL#            bsdict['shell_map'] = basisset.export_for_libmints("CABS")
-#TRIAL#            return bsdict
+    @staticmethod
+    def pyconstruct_combined(mol, keys, targets, fitroles, others):
+
+        # make sure the lengths are all the same
+        if len(keys) != len(targets) or len(keys) != len(fitroles):
+            raise ValidationError("""Lengths of keys, targets, and fitroles must be equal""")
+
+        # Create (if necessary) and update qcdb.Molecule
+        if isinstance(mol, str):
+            mol = Molecule(mol)
+            returnBasisSet = False
+        elif isinstance(mol, Molecule):
+            returnBasisSet = True
+        else:
+            raise ValidationError("""Argument mol must be psi4string or qcdb.Molecule""")
+        mol.update_geometry()
+
+        # load in the basis sets
+        sets = []
+        name = ""
+        keywords = ""
+        blends = ""
+        for at in range(len(keys)):
+            bas = BasisSet.pyconstruct(mol, keys[at], targets[at], fitroles[at], others[at])
+            name += targets[at] + " + "
+            keywords += keys[at] + " + "
+            blends += bas.name.upper() + " + "
+            sets.append(bas)
+
+        name = name[:-3].strip()
+        keywords = keywords[:-3].strip()
+        blends = blends[:-3].strip()
```

Comment:
```suggestion
        sets = [BasisSet.pyconstruct(mol, keys[at], targets[at], fitroles[at], others[at]) for at in range(len(keys))]
        name = " + ".join(targets)
        keywords = " + ".join(keys)
        blends = " + ".join([bas.name.upper() for bas in sets])
```
something like this could also work.